### DISCUSSION TITLE
How does Pyinstaller bundles resources?

### DISCUSSION THREAD

**Discussion by Blank-c** [pyinstaller/pyinstaller#7667] [PyInstaller] [answered]
As far as I know, Pyinstaller comes with prebuilt bootloader (and source code if you want to recompile it). But one cannot embed resources directly into a compiled binary. How does it creates a single exe file then? 

PS: I am not much familiar with C/C++ so it isn't like I am totally confident with what I said. Please correct me if I am wrong at any place.

**Comment by rokm** [comment, marked as answer]
On Windows, we just append the generated embedded archive at the end of the executable.

> But one cannot embed resources directly into a compiled binary.

There's win32 API that allows you to modify resources on existing executables: https://learn.microsoft.com/en-us/windows/win32/menurc/using-resources. We don't use that for embedded archive (probably due to size limitations), but we do for some other stuff, like icons, version information, and embedded manifest ([here](https://github.com/pyinstaller/pyinstaller/blob/1fa262dc36d289c0c7a3049c2772d02c31806497/PyInstaller/building/api.py#L677-L747)).

**Comment by Blank-c** [comment]
> On Windows, we just append the generated embedded archive at the end of the executable.

That makes sense thanks.

Are you planning to make any changes to this method in the near future?

**Reply by rokm** [reply]
> > On Windows, we just append the generated embedded archive at the end of the executable.
> 
> That makes sense thanks.
> 
> Are you planning to make any changes to this method in the near future?

Nope.
